src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -4364,43 +4364,43 @@ s32 func_80838FB8(PlayState* play, Player* this) {
  * Note: grottos and normal fairy fountains use `ENTR_RETURN_GROTTO`
  */
 s16 sReturnEntranceGroupData[] = {
-    // ENTR_RETURN_DAIYOUSEI_IZUMI
-    /*  0 */ ENTR_DEATH_MOUNTAIN_TRAIL_4,  // DMT from Magic Fairy Fountain
-    /*  1 */ ENTR_DEATH_MOUNTAIN_CRATER_3, // DMC from Double Defense Fairy Fountain
-    /*  2 */ ENTR_HYRULE_CASTLE_2,         // Hyrule Castle from Dins Fire Fairy Fountain
+    // ENTR_RETURN_GREAT_FAIRYS_FOUNTAIN_MAGIC
+    /*  0 */ ENTR_DEATH_MOUNTAIN_TRAIL_4,  // from Magic Fairy Fountain
+    /*  1 */ ENTR_DEATH_MOUNTAIN_CRATER_3, // from Double Defense Fairy Fountain
+    /*  2 */ ENTR_HYRULE_CASTLE_2,         // from Double Magic Fairy Fountain (as adult)
```

Comment:
This looks swapped. Double magic is in Death Mountain Crater, while Double Defence is outside Ganon's Castle (shared with Hyrule Castle)

Reply:
Indeed, will fix